convert2HMI.py

```diff
@@ -1,169 +1,208 @@
-import os
-import sys
-
 import argparse
-import yaml
 import logging
+from itertools import chain
+from pathlib import Path
 
 import numpy as np
-from astropy.io import fits
-import astropy.units as u
-import sunpy.map
-
 import torch
+import yaml
 from torch.utils.data.dataloader import DataLoader
 
 from source.models.model_manager import BaseScaler
-from source.dataset import FitsFileDataset
-from source.data_utils import get_array_radius, get_image_from_array, plot_magnetogram
+from source.utils import get_logger
 
-def get_logger(name):
+
+def get_config(instrument, fulldisk, zero_outside, add_noise, no_rescale, **kwargs):
     """
-    Return a logger for current module
+    Get config object setting values passed.
+
+    Parameters
+    ----------
+    instrument : str
+        Instrument name
+    fulldisk : bool
+        Fulldisk based inference, default is patch based
+    zero_outside :
+        Set region outside solar radius to zeros instead of default nan values
+    add_noise : float (optional)
+        Scale or standard deviation of noise to add
+    no_rescale : bool (optional)
+        Disable rescaling
+
     Returns
     -------
-
-    logger : logger instance
-
+    tuple
+        Run and config dict
     """
-    logger = logging.getLogger(name)
-    logger.setLevel(logging.DEBUG)
-    formatter = logging.Formatter(fmt="%(asctime)s %(levelname)s %(name)s: %(message)s",
-                                  datefmt="%Y-%m-%d - %H:%M:%S")
-    console = logging.StreamHandler(sys.stdout)
-    console.setLevel(logging.DEBUG)
-    console.setFormatter(formatter)
-
-    logfile = logging.FileHandler('run.log', 'w')
-    logfile.setLevel(logging.DEBUG)
-    logfile.setFormatter(formatter)
-
-    logger.addHandler(console)
-    logger.addHandler(logfile)
-
-    return logger
-
-if __name__ == '__main__':
-    logger = get_logger(__name__)
-    parser = argparse.ArgumentParser()
-    parser.add_argument('--instrument', required=True)
-    parser.add_argument('--data_path')
-    parser.add_argument('--destination')
-    parser.add_argument('--add_noise')
-    parser.add_argument('--plot', action='store_true')
-    parser.add_argument('--overwrite', action='store_true')
-    parser.add_argument('--use_patches', action='store_true')
-    parser.add_argument('--zero_outside', action='store_true')
-    parser.add_argument('--no_rescale', action='store_true')
-
-
-    args = parser.parse_args()
-    instrument = args.instrument.lower()
-
     if instrument == 'mdi':
-        run = 'checkpoints/mdi/20200312194454_HighResNet_RPRCDO_SSIMGradHistLoss_mdi_19'
+        run_dir = Path('checkpoints/mdi/20200312194454_HighResNet_RPRCDO_SSIMGradHistLoss_mdi_19')
```

Comment:
Should alway try to use `pathlib.Path` for all paths in general also if we want this work on window is is a must.